src/Orchard.Web/Modules/Orchard.Widgets/Services/DefaultLayerEvaluationService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using Orchard.Localization;
+using Orchard.Logging;
+using Orchard.Widgets.Models;
+using Orchard.ContentManagement;
+using Orchard.Core.Common.Utilities;
+
+namespace Orchard.Widgets.Services{
+    public class DefaultLayerEvaluationService : ILayerEvaluationService {
+        private readonly IRuleManager _ruleManager;
+        private readonly IOrchardServices _orchardServices;
+
+        private readonly LazyField<IEnumerable<int>> _activeLayerIDs; 
+
+        public DefaultLayerEvaluationService(IRuleManager ruleManager, IOrchardServices orchardServices) {
+            _ruleManager = ruleManager;
+            _orchardServices = orchardServices;
+
+            Logger = NullLogger.Instance;
+            T = NullLocalizer.Instance;
+
+            _activeLayerIDs = new LazyField<IEnumerable<int>>();
+            _activeLayerIDs.Loader(PopulateActiveLayers);
+        }
+
+        public ILogger Logger { get; set; }
+        public Localizer T { get; private set; }
+
+        /// <summary>
+        /// Retrieves every Layer from the Content Manager and evaluates each one.
+        /// </summary>
+        /// <returns>
+        /// A collection of integers that represents the Ids of each active Layer
+        /// </returns>
+        public IEnumerable<int> GetActiveLayerIds() {
+            return _activeLayerIDs.Value;
+        }
+
+        private IEnumerable<int> PopulateActiveLayers() {
+            // Once the Rule Engine is done:
+            // Get Layers and filter by zone and rule
+            // NOTE: .ForType("Layer") is faster than .Query<LayerPart, LayerPartRecord>()
+            var activeLayers = _orchardServices.ContentManager.Query<LayerPart>().ForType("Layer").List();
```

Comment:
TIP: Don't assume a query is faster than another, unless you have tried with a database containing millions of content items.